samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -142,13 +148,15 @@ public void onBecomeLeader() {
 
   @Override
   public void onProcessorChange(List<String> processorIds) {
-    log.info("ZkJobCoordinator::onProcessorChange - Processors changed! List: " + Arrays.toString(processorIds.toArray()));
+    log.info("ZkJobCoordinator::onProcessorChange - Processors changed! List size=" + processorIds.size());
+    // if processorIds is empty - it means we are called from 'onBecomeLeader'
+    //We currently ignore the list: " + Arrays.toString(processorIds.toArray()
```

Comment:
Disregarding the list returned from ZK event notification and fetching it again can cause inconsistency of the state in the job coordinator's view. The correct approach would be to pass this list into generateNewJobModel and fetch data only for this list. That way, the view provided by Zk and acted upon by the JC (the event handler) will be the same.

For the initial case where generateNewJobModel is called by the newly elected leader, this list can be empty and the leader can poll for the current list. This scenario can be differentiated using a flag argument in generateNewJobModel method that will indicate whether to ignore the provided list or not.